### ISSUE TITLE
CUSA06093 - Hatsune Miku: Project DIVA Future Tone

### ISSUE THREAD

**Issue by StevenMiller123** [shadps4-emu/shadps4-game-compatibility#48]
### Game Name

Hatsune Miku: Project DIVA Future Tone

### Game code

CUSA06093

### Game version

v1.00

### Used emulator's version (only released versions are acceptable)

v0.1.0

### Current status

Nothing

### Error

Without libSceNgs2.sprx, the game crashes with "[Core] <Error> stubs.cpp:CommonStub:42: Stub: sceNgs2RackGetVoiceHandle (nid: MwmHz8pAdAo) called, returning zero to 0x90011d06e"

Adding libSceNgs2.sprx to the user/sys_modules folder fixes this error, and the game will instead crash after "[Lib.AudioOut] <Error> audioout.cpp:sceAudioOutOpen:277: Invalid format"

I've provided logs for both scenarios below.

### Description

The game crashes instantly.

### Log File

[Hatsune Miku Project DIVA Future Tone Log 1.txt](https://github.com/user-attachments/files/16075412/Hatsune.Miku.Project.DIVA.Future.Tone.Log.1.txt)
[Hatsune Miku Project DIVA Future Tone Log 2.txt](https://github.com/user-attachments/files/16075413/Hatsune.Miku.Project.DIVA.Future.Tone.Log.2.txt)

**Comment by StevenMiller123** [comment]
With emulator v0.2.0, the game now reaches menus, although the menus are incorrectly rendered. Attempting to load the game causes the game to spam 
`[Lib.SaveDataDialog] <Error> savedatadialog.cpp:sceSaveDataDialogIsReadyToDisplay:32: (STUBBED) called`
This appears to be because of the return value of sceSaveDataDialogIsReadyToDisplay, as changing the return value progresses the game further. 

![Screenshot 2024-08-15 142541](https://github.com/user-attachments/assets/27706e11-1e47-490f-9959-959e3a44946b)

For now, the game's status should be changed to Menus.
Log: [Hatsune Miku Project Diva Future Tone log.txt](https://github.com/user-attachments/files/16629468/Hatsune.Miku.Project.Diva.Future.Tone.log.txt)

**Comment by viniciuslrangel** [comment]
UI elements were fixed
![image](https://github.com/user-attachments/assets/83e78bc8-f030-4da2-8d9e-797ed4e7d4f8)

**Comment by DeadBossKill** [comment]
In my case the game doesn't boot, it simply closes without displaying any error messages. 
![Sem título](https://github.com/user-attachments/assets/c658b010-48ed-4752-b2c5-f410e6ba514e)
I can't take prints properly because it goes too fast

**Comment by viniciuslrangel** [comment]
Playable! But still crashing in some machines

![image](https://github.com/user-attachments/assets/58236b5f-9da9-4a39-a511-60e46b018d21)
![image](https://github.com/user-attachments/assets/cb6c8408-4643-4b39-a5ef-2218e6ae42a4)

**Comment by StevenMiller123** [comment]
On v0.4.0, the game behaves identically to v0.3.0, with some AMD cards crashing on the following error when trying to go ingame. 
```
[Debug] <Critical> vk_scheduler.cpp:operator():147: Assertion Failed!
Device lost during submit
```
The game remains playable on Intel and Nvidia GPUs.
[shad_log.txt](https://github.com/user-attachments/files/17592132/shad_log.txt)

**Comment by DeadBossKill** [comment]
I tried to open it several times with or without modules, without success, here are the prints of the cmd in the attempts

![image](https://github.com/user-attachments/assets/b5fb2053-9477-448b-8ce8-69f98a1d1acf)

![image](https://github.com/user-attachments/assets/d4652c69-9d88-41fc-a966-3bd02b83ea69)

![image](https://github.com/user-attachments/assets/1e2dfe82-55c3-4324-8216-0154a40082f5)

![image](https://github.com/user-attachments/assets/0f493e69-68a5-4929-903c-f2865128a857)

game still won't open

PC: Xeon 2620v3
Video card: RTX 2060 6gb
16 GB DDR4
OS: Windows 10

**Comment by KaylaD23** [comment]
I have managed to sort of get ingame i can justy play the tutorial song over and over and thats it because i get a white screen when skipping the tutorial or finishing it and it will stay like that infinitely

**Comment by jrenatus** [comment]
I have the same problem mentioned  by [KaylaD23](https://github.com/KaylaD23)

**Comment by StevenMiller123** [comment]
Ingame on AMD, same device lost crash as my prior reports (though for whatever reason, the actual crash never got logged). 
![image](https://github.com/user-attachments/assets/d81c6a18-8da4-416c-93a5-fbf41eb2995a)
[shad_log.txt](https://github.com/user-attachments/files/18247664/shad_log.txt)

Probably still fully playable for other GPU vendors, but I don't have the time to test those at the moment.

**Comment by Hermiten** [comment]
Ok, I put ingame label but not regression because depends of the gpu

**Comment by systemr1982** [comment]
I can't make a screenshot of it because the whole pc hangs when this happens, but with my AMD 7840/AMD RADEON 780M, when starting any song, or opening the editing screen for modules the game crashes after running through some shaders and then (100% of the cases) crashes with a shader (permutation). The music keeps going in the background for maybe half a minute while the whole pc is unresponsive during that time and then shadps4.exe crashes. I also tried it with the Japanese DX version of the game which is categorised as playable, but it gives me the same crash every single time. https://github.com/shadps4-emu/shadps4-game-compatibility/issues/2300.

Playing on Windows.

**Comment by StevenMiller123** [comment]
@systemr1982 That would be the same issue I was reporting in v0.5.0, and the sole reason the game is still marked status-ingame instead of playable.

**Comment by systemr1982** [comment]
> [@systemr1982](https://github.com/systemr1982) That would be the same issue I was reporting in v0.5.0, and the sole reason the game is still marked status-ingame instead of playable.

Yes exactly, I just wanted to include my CPU/GPU with it so it's hopefully more clear on what kind of cards this error occurs :)

**Comment by IgnacioRelictaSoftwares** [comment]
Full playable 60fps stable with i7 12th 12650H, and Nvidia 3060RTX laptop GPU. With 16GB RAM at ShadPS4 0.6.0.
